server/src/test/java/org/elasticsearch/search/runtime/StringScriptFieldFuzzyQueryTests.java

```diff
@@ -68,18 +75,42 @@ protected StringScriptFieldFuzzyQuery mutate(StringScriptFieldFuzzyQuery orig) {
     @Override
     public void testMatches() {
         StringScriptFieldFuzzyQuery query = StringScriptFieldFuzzyQuery.build(randomScript(), leafFactory, "test", "foo", 1, 0, false);
-        assertTrue(query.matches(List.of("foo")));
-        assertTrue(query.matches(List.of("foa")));
-        assertTrue(query.matches(List.of("foo", "bar")));
-        assertFalse(query.matches(List.of("bar")));
+        BytesRefBuilder scratch = new BytesRefBuilder();
+        assertTrue(query.matches(List.of("foo"), scratch));
+        assertTrue(query.matches(List.of("foa"), scratch));
+        assertTrue(query.matches(List.of("foo", "bar"), scratch));
+        assertFalse(query.matches(List.of("bar"), scratch));
         query = StringScriptFieldFuzzyQuery.build(randomScript(), leafFactory, "test", "foo", 0, 0, false);
-        assertTrue(query.matches(List.of("foo")));
-        assertFalse(query.matches(List.of("foa")));
+        assertTrue(query.matches(List.of("foo"), scratch));
+        assertFalse(query.matches(List.of("foa"), scratch));
         query = StringScriptFieldFuzzyQuery.build(randomScript(), leafFactory, "test", "foo", 2, 0, false);
-        assertTrue(query.matches(List.of("foo")));
-        assertTrue(query.matches(List.of("foa")));
-        assertTrue(query.matches(List.of("faa")));
-        assertFalse(query.matches(List.of("faaa")));
+        assertTrue(query.matches(List.of("foo"), scratch));
+        assertTrue(query.matches(List.of("foa"), scratch));
+        assertTrue(query.matches(List.of("faa"), scratch));
+        assertFalse(query.matches(List.of("faaa"), scratch));
+    }
+
+    public void testConcurrentMatches() {
```

Comment:
These concurrency test are kind of artificial now that they rely on the different matches method... we could rely on integration tests perhaps instead.

Reply:
If you built a `Scorer` or something that could work. That's kind of integration-y. You'd need to make a lucene index, but sometimes that's life.

Reply:
Yep, I later realized that we already have coverage for all of this in our script field type tests. Only, those did not leverage concurrency as we never ended up running them against multiple segments. I have added an addDocument method to the base class that more aggressively flushes, this is ok as we are only ever adding a few documents. The random flush within RandomIndexWriter needs a minimum number of 10 docs which is never reached in these tests.
With this adjustment, I was able to reproduce the problem and ensure that it is now fixed.